bluesky_widgets/examples/kafka_figures.py

```diff
@@ -0,0 +1,68 @@
+"""
+Run like:
+
+python -m bluesky_widgets.examples.kafka_figures
+
+It will save a png file to ~/bluesky_widgets_test/{run_uid} for each run.
+"""
+from pathlib import Path
+from functools import partial
+import os
+
+import msgpack
+import msgpack_numpy as mpn
+from bluesky_kafka import RemoteDispatcher
+
+from bluesky_widgets.utils.streaming import stream_documents_into_runs
+from bluesky_widgets.models.plot_builders import AutoRecentLines
+from bluesky_widgets.headless.figures import HeadlessFigures
+from bluesky_widgets.examples.utils.generate_msgpack_data import get_catalog
+from bluesky_widgets.models.utils import run_is_live_and_not_completed
+
+
+def export_thumbnails_when_complete(run):
+    "Given a BlueskyRun, export thumbnail(s) to a directory when it completes."
+    model = AutoRecentLines(3)
+    model.add_run(run)
+    view = HeadlessFigures(model.figures)
+
+    home = str(Path.home())
+    base_directory = os.path.join(home, "bluesky_widgets_test")
+    if not os.path.exists(base_directory):
+        os.makedirs(base_directory)
+    uid = run.metadata['start']['uid']
+    directory = os.path.join(base_directory, uid)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
```

Comment:
`os.makedirs(directory, exist_ok=True)` will let you avoid the `if ...`. Also, You can make nested directories this way ("make dirS") so no need to make the `base_directory` separately.